plugins/terraform/resource_load_balancer_test.go

```diff
@@ -0,0 +1,250 @@
+package main
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/quintilesims/layer0/common/models"
+	"testing"
+)
+
+func TestLoadBalancerCreate_basic(t *testing.T) {
+	ctrl, mockClient, provider := setupUnitTest(t)
+	defer ctrl.Finish()
+
+	loadBalancerResource := provider.ResourcesMap["layer0_load_balancer"]
+	d := schema.TestResourceDataRaw(t, loadBalancerResource.Schema, map[string]interface{}{
+		"name":        "test-lb",
+		"environment": "test-env",
+	})
+
+	mockClient.EXPECT().
+		CreateLoadBalancer("test-lb", "test-env", models.HealthCheck{"TCP:80", 30, 5, 2, 2}, []models.Port{}, true).
```

Comment:
Will the layer0 api fail this call if we send an empty `[]models.Port{}`? I don't recall the layer0 api setting a default.